src/core/manager.c

```diff
@@ -117,9 +117,19 @@ static int manager_run_generators(Manager *m);
 static void manager_vacuum(Manager *m);
 
 static usec_t manager_watch_jobs_next_time(Manager *m) {
-        return usec_add(now(CLOCK_MONOTONIC),
-                        show_status_on(m->show_status) ? JOBS_IN_PROGRESS_WAIT_USEC :
-                                                         JOBS_IN_PROGRESS_QUIET_WAIT_USEC);
+        usec_t timeout;
+
+        if (show_status_on(m->show_status))
+                /* When status is on, just use the usual timeout. */
+                timeout = JOBS_IN_PROGRESS_WAIT_USEC;
+        else if (MANAGER_IS_SYSTEM(m))
+                timeout = JOBS_IN_PROGRESS_QUIET_WAIT_USEC;
+        else
+                /* Let the user manager without a timeout show status quickly, so the system manager can make
+                 * use of it, if it wants to. */
+                timeout = JOBS_IN_PROGRESS_WAIT_USEC * 2 / 3;
```

Comment:
hmm, the order of these checks is a bit surprising? i.e. the show_status_on() stuff is pointless to check in the user instance. Hence it appear to me, you want this last else block to be put first, conditionalized by MANAGER_IS_USER(), and then put the other two branches in its else. i.e.:

```c
if (MANAGER_IS_USER(m)) 
        timeout = JOBS_IN_PROGRESS_WAIT_USEC * 2 / 3;
else if (show_status_on(…))
        timeout = …
else
        timeout = …
```

Reply:
Yeah, that's probably better.